docs/reference/ml/df-analytics/apis/delete-trained-model.asciidoc

```diff
@@ -0,0 +1,59 @@
+[role="xpack"]
+[testenv="basic"]
+[[delete-inference]]
+=== Delete trained model API
```

Comment:
Related to Ben's comment (https://github.com/elastic/elasticsearch/pull/50224#discussion_r358200945), it caught my eye that the title is "delete trained model API", but the URL is "delete-inference.html" and the intro is "deletes an existing trained inference model".
I think we need to make the three match (i.e. either we're explicitly mentioning inference in the title, url, and intro or we're not).